### PR TITLE
Test cases added

### PR DESCRIPTION
**Pull request by debasen**
### Counter

- Test that component renders initial value provided in props
- Test that a click event on "decrement" button decrements the displayed value
- Test that a click event on "increment" button increments the displayed value

### SearchForm

- Test that component renders an input with the value equal to initial value passed in props
- Test that after typing to the input and a "click" event on the Submit button, the "onChange" prop is called with proper value
- Test that after typing to the input and pressing Enter key, the "onChange" prop is called with proper value

### GenreSelect

- Test that component renders all genres passed in props
- Test that component highlights a selected genre passed in props
- Test that after a click event on a genre button component calls "onChange" callback and passes correct genre in arguments

### E2E Test case using Cypress:
1. Load page
2. Click on Horror Genre
3. Selected genre should have "Horror"
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+    console.log("Selected Genre: " + selectedGenre);
+  }
   return (
     <div>
-    <SearchForm/>
-    <GenreSelect/>
-    <Counter/>
+      <SearchForm initialSearchTerm='Movie' onChange={handleSearch} />
+      <GenreSelect genres={genres} selectedGenre={selectedGenre} onChange={handleGenreSelect} />
+      <Counter initialCount={3} />
```
Comment (melvinvicky46) [inline src/App.js:18]: No magic number please, use a constant file or declare a variable and initialize.

File: src/components/searchForm/searchForm.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import '@testing-library/jest-dom'
+
+import * as React from 'react'
+import { fireEvent, render, screen, debug } from '@testing-library/react'
+import SearchForm from './searchForm';
+const searchCB = jest.fn();
```
Comment (melvinvicky46) [inline src/components/searchForm/searchForm.test.js:6]: Leave a empty line after imports

File: src/components/searchForm/searchForm.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const setup = (initialValue) => {
+    const utils = render(<SearchForm initialSearchTerm={initialValue} onChange={searchCB}></SearchForm>);
+    const searchInput = screen.getByLabelText('search-box');
+    return { ...utils, searchInput };
+}
+test('check if the search box is rendered with inital value', () => {
```
Comment (melvinvicky46) [inline src/components/searchForm/searchForm.test.js:12]: Add all the test within a describe block.

File: src/App.js (hunk 1 above)
Comment (debasen) [reply src/App.js]: Makes sense. Moved to a constant file.

File: src/components/searchForm/searchForm.test.js (hunk 2 above)
Comment (debasen) [reply src/components/searchForm/searchForm.test.js:7]: Done

File: src/components/searchForm/searchForm.test.js (hunk 3 above)
Comment (debasen) [reply src/components/searchForm/searchForm.test.js]: Wrapped tests inside `describe` block.

**Review by melvinvicky46 — APPROVED** [review]

**Pull request merged**